packages/core/src/shared/utilities/cacheUtils.ts

```diff
@@ -131,17 +130,16 @@ export function createDiskCache<T, K>(
                 await fs.mkdir(dirname(target))
                 if (isWeb()) {
                     // There is no web-compatible rename() method. So do a regular write.
-                    await fs.writeFile(target, JSON.stringify(data), { mode: 0o600 })
+                    await fs.writeFile(target, JSON.stringify(data))
                 } else {
                     // With SSO cache we noticed malformed JSON on read. A guess is that multiple writes
                     // are occuring at the same time. The following is a bandaid that ensures an all-or-nothing
                     // write, though there can still be race conditions with which version remains after overwrites.
-                    const tempFile = `${target}.tmp-${crypto.randomBytes(4).toString('hex')}`
-                    await fs.writeFile(tempFile, JSON.stringify(data), { mode: 0o600 })
-                    await fs.rename(tempFile, target)
+                    await fs.writeFile(target, JSON.stringify(data), { atomic: true })
                 }
             } catch (error) {
-                throw ToolkitError.chain(error, `Failed to save "${target}"`, {
+                const message = error instanceof Error ? error.message : 'NoMessage'
+                throw ToolkitError.chain(error, `Failed to save "${target}" with message: ${message}`, {
```

Comment:
message is contained in the `cause` already, which is done by chain(). why do we need to repeat it here?

if nested messages should be shown to the user, then we should improve our error handling rather than duplicating messages in each part of the chain.

Reply:
In telemetry we do not show the nested error in `reasonDesc`. So in this example before my change we only saw: `Failed to save "${target}"`, the message from the wrapped error was lost.

Reply:
Opened a PR to address the underlying issue: https://github.com/aws/aws-toolkit-vscode/pull/5373